integration/kubernetes/helm-chart/alluxio/templates/worker/daemonset.yaml

```diff
@@ -68,16 +68,13 @@ spec:
       {{- if .Values.tolerations }}
 {{ toYaml .Values.tolerations | trim | indent 8  }}
       {{- end }}
-    {{- if eq .Values.worker.fuseEnabled true }}
-      securityContext:
-        runAsUser: {{ .Values.fuse.user }}
-        runAsGroup: {{ .Values.fuse.group }}
-        fsGroup: {{ .Values.fuse.fsGroup }}
-    {{- else }}
       securityContext:
-        runAsUser: {{ .Values.user }}
-        runAsGroup: {{ .Values.group }}
-        fsGroup: {{ .Values.fsGroup }}
+        runAsUser: {{ .Values.worker.user }}
+        runAsGroup: {{ .Values.worker.group }}
+        fsGroup: {{ .Values.worker.fsGroup }}
+    {{- if eq .Values.worker.fuseEnabled true }}
+    {{- if or (or (ne .Values.worker.user 0) (ne .Values.worker.group 0)) (ne .Values.worker.fsGroup 0) }}
+    {{- fail ".Values.worker.{user, group, fsGroup} have to be 0 in order to launch Fuse in worker process" }}
```

Comment:
Doesn't this implementation prevent the worker pods from using the deployment-wide defaults in `.Values.(user|group|fsGroup)`?

I'd prefer for `.Values.worker.(user|group|fsGroup)` to override `.Values.(user|group|fsGroup)`, and have the default in `values.yaml` be empty or some other way to indicate to use the global values. Eg:

```
worker:
  user:    # supply a value to override the top-level defaults
  group:   # supply a value to override the top-level defaults
  fsGroup: # supply a value to override the top-level defaults
```

```
securityContext:
  {{- if .Values.worker.user }}
  runAsUser: {{ .Values.worker.user }}
  {{- else }}
  runAsUser: {{ .Values.user }}
  {{- end }}
```

You'd also need to change your implementation of checking that the assigned securityContext is running with UID/GID `0` too.

Reply:
Thanks for the clarification! I misunderstood what you meant yesterday.